code-challenges.test.js

```diff
@@ -14,33 +14,329 @@
+  
+  // a) Create a test with an expect statement using the variables provided.
+  
+  // Pseudo Code:
+    // Input: Array
+    // Output: Number
+    // Process:
+      // Function Name: cubedSum 
+      // Assign parameter name 'array' to the function
+      // Create temporary number 'tempNum'
+      // Iterate through the array (possible tool: .map)
+        // Cube the current value
+        // Add the cubed current value to the temporary number
+      // Return the final count
+    // Expected Output: 99
+                     // 1125
+  
+  
+  describe("cubedSum", () => {
+    
+    const cubeAndSum1 = [2, 3, 4]
+    // Expected output: 99
+    const cubeAndSum2 = [0, 5, 10]
+    // Expected output: 1125
+    
+    it("takes in an array of numbers and returns the sum of all the numbers cubed.", () => {
+      expect(cubedSum(cubeAndSum1)).toEqual(99)
+      expect(cubedSum(cubeAndSum2)).toEqual(1125)
+    })
+  })
+  
+  // Good failure to due cubedSum() not defined
+  // FAIL  ./challenges.test.js
+  // createSent
+  //   ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (1 ms)
+  // remainGen
+  //   ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3. (1 ms)
+  // cubedSum
+  //   ✕ takes in an array of numbers and returns the sum of all the numbers cubed.
+  
+  // ● cubedSum › takes in an array of numbers and returns the sum of all the numbers cubed.
+  
+  //   ReferenceError: cubedSum is not defined
+  
+  // b) Create the function that makes the test pass.
+  
+  // const cubedSum = (array) => {
+  
+  //   let tempNum = 0
+  
+  //   // Iterates through 'array'
+  //   array.map(value => {
+  //     // Computes the current value to the power of 3
+  //     // Add the resulting value to 'tempNum'
+  //     tempNum += Math.pow(value, 3)
+  //   })
+  
+  //   return tempNum
+  
+  // }
+  
+  // Success
+  // PASS  ./challenges.test.js
+  // createSent
+  //   ✓ takes in an array of objects and returns an array with a sentence about each person with their name capitalized. (1 ms)
+  // remainGen
+  //   ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3. (1 ms)
+  // cubedSum
+  //   ✓ takes in an array of numbers and returns the sum of all the numbers cubed.
+  
+  // Refactor 
+  // Refactor 
+  // Pseudo Code:
+    // Process:
+      // Remove 'tempNum'
+      // Remove .map
+      // Iterate through the array with .reduce()
+      // Add the values to the power of 3
+    // Expected Output: 99
+                     // 1125
+  
+  const cubedSum = (array) => {
+  
+    // Iterate through the array with .reduce()
+      // 'a' is the accumulated value
+      // 'b' is the next number to be added
+    // Set initialized accumulated value to 0
+    // Adds 'a' to Math.pow function
+    return array.reduce((a,b) => a + Math.pow(b, 3), 0)
```

Comment:
Nicely done, you certainly have a good understanding of .reduce. Great use of it